src/views/Dashboard/Customer/Home/MyTest/TestResult.vue

```diff
@@ -130,19 +269,52 @@ export default {
       this.showModal = false
     },
 
-    submitRating() {
-      this.showModalRating = false
-      this.showModal = true
+    getRating(stars) {
+      this.rating = stars
+    },
+
+    async submitRating() {
+      try {
+        await submitRatingOrder(
+          this.speciment.lab_id,
+          this.order.service_id,
+          this.speciment.order_id,
+          this.order.customer_id,
+          this.rating
+        );
+
+        this.showModalRating = false
+        this.showModal = true
+      } catch (error) {
+        this.showModalRating = false
+        this.showModal = true
+      }
     }
   },
 
   computed: {
+    ...mapState({
+      api: (state) => state.substrate.api,
+      wallet: (state) => state.substrate.wallet,
+      mnemonicData: (state) => state.substrate.mnemonicData
+    }),
+
+    reportResult() {
+      if (this.dialog) {
+        return "";
+      }
+
+      if (this.resultLoading) {
+        return "Decrypting report..";
+      }
```

Comment:
please remove redundant lines, your if statement can be minimalist by doing this

```js
if (something) doSomething
```

Reply:
No, this shouldn't be a requirement, that's very subjective. I can argue:

```
if (something) {
  doSomething
}
```

Makes things more readable.

Reply:
Of course, I agree with you @agustinustheo, this is very subjective and you are free to choose, actually I'm just suggesting. Because on many of the projects I worked on in the past, which were also where they carried the "Clean Code" concept they highly recommended this method. So it's up to you to use it or not. So feel free to choose any method. 😉